grakn-graph/src/test/java/ai/grakn/generator/GraknGraphs.java

```diff
@@ -147,30 +147,40 @@ public GraknGraphs setOpen(boolean open) {
     private final ImmutableList<Runnable> mutators = ImmutableList.of(
             () -> {
                 TypeName typeName = typeName();
-                EntityType entityType = graph.putEntityType(typeName);
+                EntityType superType = entityType();
+                EntityType entityType = graph.putEntityType(typeName).superType(superType);
```

Comment:
This can fail if by some miracle `typeName == superType.getName()`. Also fails if meta type are involved. Actually what happens if this does fail?